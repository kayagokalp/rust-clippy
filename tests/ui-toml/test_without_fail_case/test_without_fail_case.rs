//@aux-build:test_macro.rs
//@compile-flags: --test
//@revisions: default fail_macro no_fail_macro index_fail
//@[default] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/test_without_fail_case/default
//@[fail_macro] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/test_without_fail_case/fail_macro
//@[no_fail_macro] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/test_without_fail_case/no_fail_macro
//@[index_fail] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/test_without_fail_case/index_fail
#![allow(unused)]
#![allow(clippy::tests_outside_test_module)]
#![warn(clippy::test_without_fail_case)]
use test_macro::{fallible_macro, non_fallible_macro};

// Should not lint because of the clippy.toml file that adds `test` as fallible.
#[test]
fn test_custom_macro_fallible() {
    println!("a")
}

// Should not lint unless the clippy.toml file makes indexing fallible.
#[test]
fn test_indexing_fallible() {
    let a = [1, 2, 3];
    let _ = a[0];
}

#[test]
fn func_a() {
    let _ = 1;
}

#[test]
fn should_not_lint_if_defined_as_fallible() {
    non_fallible_macro!(1);
}

#[test]
fn should_lint_if_defined_as_non_fallible() {
    fallible_macro!(1);
}

fn main() {}
