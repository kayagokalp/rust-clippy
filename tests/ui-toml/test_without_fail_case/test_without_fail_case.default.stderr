error: this function marked with `#[test]` cannot fail and will always succeed
  --> tests/ui-toml/test_without_fail_case/test_without_fail_case.rs:15:1
   |
LL | / fn test_custom_macro_fallible() {
LL | |     println!("a")
LL | | }
   | |_^
   |
   = note: consider adding assertions or panics to test failure cases
   = note: `-D clippy::test-without-fail-case` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::test_without_fail_case)]`

error: this function marked with `#[test]` cannot fail and will always succeed
  --> tests/ui-toml/test_without_fail_case/test_without_fail_case.rs:21:1
   |
LL | / fn test_indexing_fallible() {
LL | |     let a = [1, 2, 3];
LL | |     let _ = a[0];
LL | | }
   | |_^
   |
   = note: consider adding assertions or panics to test failure cases

error: this function marked with `#[test]` cannot fail and will always succeed
  --> tests/ui-toml/test_without_fail_case/test_without_fail_case.rs:27:1
   |
LL | / fn func_a() {
LL | |     let _ = 1;
LL | | }
   | |_^
   |
   = note: consider adding assertions or panics to test failure cases

error: this function marked with `#[test]` cannot fail and will always succeed
  --> tests/ui-toml/test_without_fail_case/test_without_fail_case.rs:32:1
   |
LL | / fn should_not_lint_if_defined_as_fallible() {
LL | |     non_fallible_macro!(1);
LL | | }
   | |_^
   |
   = note: consider adding assertions or panics to test failure cases

error: aborting due to 4 previous errors

